cmd/img_mgmt/src/img_mgmt.c

```diff
@@ -165,7 +165,15 @@ img_mgmt_read_info(int image_slot, struct image_version *ver, uint8_t *hash,
      * the hash is missing, the image is considered invalid.
      */
     data_off = hdr.ih_hdr_size + hdr.ih_img_size;
-    rc = img_mgmt_find_tlvs(&hdr, image_slot, &data_off, &data_end);
+
+    rc = img_mgmt_find_tlvs(image_slot, &data_off, &data_end, IMAGE_TLV_PROT_INFO_MAGIC);
+    if (rc != 0) {
+        return MGMT_ERR_EUNKNOWN;
+    }
```

Comment:
Some images won't contain protected TLVs (e.g., older ones).  I think the code should ignore an error here and only advance the offset on success.

Reply:
done